netket/vqs/mc/mc_state/state.py

```diff
@@ -549,6 +550,50 @@ def log_value(self, σ: jnp.ndarray) -> jnp.ndarray:
         """
         return jit_evaluate(self._apply_fun, self.variables, σ)
 
+    def local_estimators(self, op: AbstractOperator, *, chunk_size: int = None):
+        r"""
+        Compute the local estimators for the operator :code:`op` (also known as local energies
+        when :code:`op` is the Hamiltonian) at the current configuration samples :code:`self.samples`.
+
+        .. math::
+
+            O_\mathrm{loc}(s) = \frac{\langle s | \mathtt{op} | \psi \rangle}{\langle s | \psi \rangle}
+
+        .. warning::
+
+            The samples differ between MPI processes, so returned the local estimators will
+            also take different values on each process. To compute sample averages and similar
+            quantities, you will need to perform explicit operations over all MPI ranks.
+            (Use functions like :code:`self.expect` to get process-independent quantities without
+            manual reductions.)
+
+        Args:
+            op: The operator.
+            chunk_size: Suggested maximum size of the chunks used in forward and backward evaluations
+                of the model. (Default: :code:`self.chunk_size`)
+        """
+        s, extra_args = get_local_kernel_arguments(self, op)
+
+        shape = s.shape
+        if jnp.ndim(s) != 2:
+            s = s.reshape((-1, shape[-1]))
+
+        def logpsi(w, s):
+            return self._apply_fun({"params": w, **self.model_state}, s)
+
+        if chunk_size is not None or self.chunk_size is not None:
+            if chunk_size is None:
+                chunk_size = self.chunk_size
```

Comment:
I'd remove support for custom chunk_size.
If eventually we decide to keep it, I'd change this to the more easy to follow (unless this won't work)
```suggestion
        if chunk_size is None:
            chunk_size = self.chunk_size
            
        if chunk_size is not None:
```